### PR TITLE
[4.0] Saving article cause : hits = '' is not a valid integer

### PR DESCRIPTION
**Pull request by shim-sao**
Pull Request for Issue # .

### Summary of Changes
Add default value 0 to the "hits" field in the admin form of the article.

### Testing Instructions
Create an article, duplicate it.
In this case trying to save the duplicated article cause : '' is not a valid integer on save
The field hits is blank and is not be set to a valid integer in the query

This not happen all the time. Why ?

An integer must have a default valid value if it is save in the database or be set to NULL if it is allowed in the table

### Actual result BEFORE applying this Pull Request
Saving article cause : '' is not a valid integer

### Expected result AFTER applying this Pull Request
Article saved with default hits to 0

### CODE REVIEW

**Comment by BertaOctech** [comment]
Hi.

I have not been able to reproduce the issue.

I have created 2 articles, I have visited only one of them. In this way I have one article with hits = 0 and another one with hits > 0.

In both cases, when I used "Save as Copy" a new article was created, and the message "Article saved." appeared.

I have tried in Chrome and Firefox.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on a5588af087957586af0955dd81c4b8a35474950c<br /><br />To reproduce:

Edit an article.
Go to Publishing tab.
Use browser's inspector to change the value of Hits to empty string.
Click Save button.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on a5588af087957586af0955dd81c4b8a35474950c<br /><br />Followed the procedure given by BertaOctech<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

**Comment by shim-sao** [comment]
Indeed this is the fastest way to reproduce the error.
I have only described in which case it occurred on my side ... it is not systematic as I specified.

I made the most obvious correction but for me there is clearly a problem in the form validation process on the server side.
Since the field is defined as an integer, even without a default value, it should be initialized to 0 no matter what, '' is not an integer.

Maybe there is an hidden error in the integer definition filter.

I think if this happens for this field, then all forms with none initialized integer should be reviewed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

**Comment by shim-sao** [comment]
This error is no need to be reproduce for this patch but...

I checked the category hits field and the default to 0 is present.

If you remove the default of the field 'hits' (category or article) and follow the @Quy instruction, you will get the error.
You can also enter a string instead of an integer and the string will be convert to empty string.
*Incorrect integer value: '' for column 'hits' at row 1*

**This kind of error must not happen on the server side validation because everybody can make a form like he want, specially with the API**

**The form validation does not make its job if a value is set to string for an integer with no default, the default should be 0 on all case or the form is not valid**<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30523">issues.joomla.org/tracker/joomla-cms/30523</a>.</sub>

File: administrator/components/com_content/forms/article.xml [hunk 1]
```diff
@@ -944,4 +945,3 @@
 	</fields>
 
 </form>
-
```
Comment (laoneo) [inline administrator/components/com_content/forms/article.xml]: Can you add here the empty line again?

Comment (shim-sao) [reply administrator/components/com_content/forms/article.xml]: No, my file has the space at line 948 and it doesn't propose me changes for commit so this seems to be already done. ```git merge upstream/4.0-dev```, maybe I'm wrong !?

Comment (shim-sao) [reply administrator/components/com_content/forms/article.xml]: It's clearly a github bug for me. When I look at my file (on the github page), the blank line not exists at the end of the file. If I edit it, the line exists. So, github seems to do what it want with the last blank line.

Comment (shim-sao) [reply administrator/components/com_content/forms/article.xml]: Done, I add a second blank line at the end of the file and it works 👍
Note : I didn't touch it before

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**